src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -32,31 +34,38 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
         private _googlePayAdyenV2PaymentProcessor?: GooglePayAdyenV2PaymentProcessor
     ) {}
 
-    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
-        this._methodId = options.methodId;
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
 
-        return this._googlePayPaymentProcessor.initialize(this._methodId)
-            .then(() => {
-                this._googlePayOptions = this._getGooglePayOptions(options);
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
 
-                if (!this._googlePayOptions) {
-                    throw new InvalidArgumentError('Unable to initialize payment because "options.googlepay" argument is not provided.');
-                }
+        try {
+            this._googlePayOptions = this._getGooglePayOptions(options);
+        } catch (error) {
+            return Promise.reject(error);
```

Comment:
I believe you don't need to do try/catch if you will simply return a rejected promise with the same caught error.